tools/cli/adminFailoverCommands.go

```diff
@@ -0,0 +1,275 @@
+
+	var runID string
+	var descResp *shared.DescribeWorkflowExecutionResponse
+	var err error
+	if c.IsSet(FlagRunID) {
+		runID = c.String(FlagRunID)
+	} else {
+		descResp, err = client.DescribeWorkflowExecution(tcCtx, failoverManager.WorkflowID, "")
+		if err != nil {
+			ErrorAndExit("Failed to get failover workflow runID", err)
+		}
+		runID = descResp.WorkflowExecutionInfo.Execution.GetRunId()
+	}
+
+	queryResult, err := client.QueryWorkflow(tcCtx, failoverManager.WorkflowID, runID, failoverManager.QueryType)
+	if err != nil {
+		ErrorAndExit("Failed to query failover workflow", err)
+	}
+	if !queryResult.HasValue() {
+		ErrorAndExit("QueryResult has no value", nil)
+	}
+
+	var result failoverManager.QueryResult
+	queryResult.Get(&result)
+
+	if descResp == nil {
+		descResp, err = client.DescribeWorkflowExecution(tcCtx, failoverManager.WorkflowID, runID)
+		if err != nil {
+			ErrorAndExit("Failed to describe workflow", err)
+		}
+	}
+	if descResp.WorkflowExecutionInfo.CloseStatus.Equals(shared.WorkflowExecutionCloseStatusTerminated) {
+		result.State = "aborted"
+	}
+	prettyPrintJSONObject(result)
+}
+
+// AdminFailoverAbort abort a failover workflow
+func AdminFailoverAbort(c *cli.Context) {
+	client := getCadenceClient(c)
+	tcCtx, cancel := newContext(c)
+	defer cancel()
+
+	reason := c.String(FlagReason)
+	if len(reason) == 0 {
+		reason = defaultAbortReason
+	}
+	runID := getRunID(c, client)
+
+	err := client.TerminateWorkflow(tcCtx, failoverManager.WorkflowID, runID, reason, nil)
+	if err != nil {
+		ErrorAndExit("Failed to abort failover workflow", err)
+	}
+
+	fmt.Println("Failover aborted")
+}
+
+// AdminFailoverRollback rollback a failover run
+func AdminFailoverRollback(c *cli.Context) {
+	client := getCadenceClient(c)
+	tcCtx, cancel := newContext(c)
+	defer cancel()
+
+	runID := getRunID(c, client)
+
+	queryResp, err := client.QueryWorkflow(tcCtx, failoverManager.WorkflowID, runID, failoverManager.QueryType)
+	if err != nil {
+		ErrorAndExit("Failed to query failover workflow", err)
+	}
+	if !queryResp.HasValue() {
+		ErrorAndExit("QueryResult has no value", nil)
+	}
+	var queryResult failoverManager.QueryResult
+	queryResp.Get(&queryResult)
+
+	params := &startParams{
+		targetCluster:                  queryResult.SourceCluster,
+		sourceCluster:                  queryResult.TargetCluster,
+		domains:                        queryResult.SuccessDomains,
+		batchFailoverSize:              c.Int(FlagFailoverBatchSize),
+		batchFailoverWaitTimeInSeconds: c.Int(FlagFailoverWaitTime),
+		failoverTimeout:                c.Int(FlagFailoverTimeout),
+	}
+	failoverStart(c, params)
+}
+
+// AdminFailoverList list failover runs
+func AdminFailoverList(c *cli.Context) {
+	c.Set(FlagWorkflowID, failoverManager.WorkflowID)
+	ListWorkflow(c)
+}
+
+func getCadenceClient(c *cli.Context) cclient.Client {
+	svcClient := cFactory.ClientFrontendClient(c)
+	return cclient.NewClient(svcClient, common.SystemLocalDomainName, &cclient.Options{})
+}
```

Comment:
nit: can we reuse the `getWorkflowClientWithOptionalDomain()` func?

Reply:
may not worth it, because c.Set(domain) will require domain flag exist, which make it not a simple switch. will leave as it. But thanks for bring it up